### PR TITLE
Upgrading to core 0.100.3

### PR DESCRIPTION
**Pull request by kneth**
@realm/java

### CODE REVIEW

File: realm/realm-jni/build.gradle [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import java.security.MessageDigest
 
-ext.coreVersion = '0.100.2'
+ext.coreVersion = '0.100.3'
 // empty or comment out this to disable hash checking
-ext.coreSha256Hash = '081907973e45da61ea6e42b0ff05aacebbc02d28afddba0cede96362c070cce7'
+    ext.coreSha256Hash = '2d6883afc3743d59d88f3e4e8caedb79eb0155a33444e50672681549c9b96793'
```
Comment (stk1m1) [inline realm/realm-jni/build.gradle]: extra spaces.

**Comment by cmelchior** [comment]
👍

**Comment by stk1m1** [comment]
👍

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,12 +1,13 @@
 ## 0.91.0
```
Comment (beeender) [inline CHANGELOG.md]: should be 0.90.1. But if we are going to merge #2741 it is fine. It will be fixed by that .

**Comment by zaki50** [comment]
:+1:

**Comment by cmelchior** [comment]
Changing my vote to 👎 . 100.3 contains breaking changes.

**Comment by dalinaum** [comment]
If it includes breaking changes, :-1:

**Pull request closed without merging**